cost-model/src/cost_model.rs

```diff
@@ -105,8 +115,49 @@ impl CostModel {
         }
     }
 
+    /// Return an estimated total cost for a transaction given its':
+    /// - `meta` - transaction meta
+    /// - `instructions` - transaction instructions
+    /// - `num_write_locks` - number of requested write locks
+    ///
+    /// This assumes ALL features are activated.
+    /// This is intended to be used as an estimate of cost ONLY for the purposes of
+    /// prioritization and packing.
```

Comment:
`estimate_cost()` essentially does same thing as `calculate_cost()`, but with a better interface. Both provide estimated cost for packing, for that purpose, it doesn't need a fully sanitized transaction if `num_write_locks` is known. How about sunset `calculate_cost()`? Maybe first step: `fn calculate_cost(tx, feature_set) -> u64 { Self::estimate_cost(tx, tx.program_iter(), feature_set) }`?

Reply:
ps. `calculate_cost_for_executed_transaction()` (is for block limits enforcing) doesn't need fully sanitized transaction either, if `num_write_locks` is known.

Then the entire cost_model can be solely depends on StaticMeta, then "estimated cost" can itself be part of StaticMeta! wdyt?

Reply:
Yeah could do it - I guess my concern was primarily right now the number of write locks is behind a feature-gate.
Because before feature is activated we count actual number of locks.
After we count requested write locks.
So we technically need to have a fully sanitized transaction (at least to point of ALT resolution and lock demotion - which are usually our last steps) until we clean up that feature.

So it felt like a foot-gun to make that simple integer be an input for the "main" entrypoints.
I think once that feature is activated its' probably fine?

Reply:
I reworked it so that all 3 main functions: `calculate_cost`, `calculate_cost_for_executed_transaction`, `estimate_cost` all have a shared implementation called for the non-vote transactions.